depends/Makefile

```diff
@@ -282,6 +282,9 @@ $(host_prefix)/share/toolchain.cmake : toolchain.cmake.in $(host_prefix)/.stamp_
             -e 's|@CPPFLAGS@|$(strip $(host_CPPFLAGS))|' \
             -e 's|@CPPFLAGS_RELEASE@|$(strip $(host_release_CPPFLAGS))|' \
             -e 's|@CPPFLAGS_DEBUG@|$(strip $(host_debug_CPPFLAGS))|' \
+            -e 's|@LDFLAGS@|$(strip $(host_LDFLAGS))|' \
```

Comment:
In bce7beaaf06b84cec5aafd5b0207b46ac70a38f7: What ended up pointing out this error? I'm suprised that all builds (regardless of Guix) weren't broken if this was missing? Or I guess up until this point, nobody has tested setting LDFLAGS in a CMake build with depends?

Reply:
Another weird observation.

https://github.com/hebasto/bitcoin/pull/129#issuecomment-2021200417 was produced on Ubuntu 23.10. But it must fail to pass `check_MACHO_sdk` in the `symbol-check.py`.

Reply:
> I'm suprised that all builds (regardless of Guix) weren't broken...

Isn't `darwin` the only host that sets linker flags in depends?

Reply:
> Isn't darwin the only host that sets linker flags in depends?

Anyone building depends for any `HOST` with `make -C depends LDFLAGS="some_flag"` would have broken up until this point?

Reply:
Everyone has a such opportunity now :)